nautobot/core/celery/backends.py

```diff
@@ -64,3 +69,16 @@ def _get_extended_properties(self, request, traceback):
             )
 
         return extended_props
+
+    def prepare_exception(self, exc, serializer=None):
+        """Overload default to explicitly sanitize the traceback message result."""
+        exc_info = super().prepare_exception(exc, serializer=serializer)
+
+        # If this is a GitCommandError, attempt to sanitize the message error.
+        # e.g. b"fatal: could not read Password for 'https://abc123@github.com': terminal prompts disabled"
+        if exc_info["exc_type"] == "GitCommandError":
+            exc_msg = list(exc_info["exc_message"])  # It starts as a tuple.
+            exc_msg[-1] = sanitize(exc_msg[-1].decode("utf-8"))  # The message will be bytes.
+            exc_info["exc_message"] = tuple(exc_msg)  # Turn it back into a tuple.
```

Comment:
This is a lot more narrowly targeted than I'd personally prefer. Can we not just sanitize every part of `exc_message` unconditionally?

Reply:
For the fetch/clone error cases, `exc_message` coming from `GitCommandError` is a list following this pattern:

```
# ([shlex.split(), retval, error_msg])
(["git", "fetch", "other", "args"], 128, b"fatal: could not read Password for 'https://abc123@github.com': terminal prompts disabled"])
```

I was trying `json.loads(sanitize(json.dumps(exc_info["exc_message"])))` to do it in one shot but that stupid error message is `bytes` for some reason.

The other thing is that I have no assertion that other errors would be in this same format, so I kept it super specific to the reported issue.

Reply:
Also I figure now that there is a pattern for sanitizing exceptions coming from Celery, that we might end up with mappings on how to do each one as-needed, unless we can come-up with a byte-steamrolling-JSON-encoder.